File: skill-test/projects/sample-cluster-dotnet/ClusterActors/HelloActor.cs
using Akka.Actor;

namespace ClusterActors;

public interface IWelcomeMessageProvider
{
    string GetMessage();
}

public sealed class WelcomeMessageProvider : IWelcomeMessageProvider
{
    public string GetMessage() => "wellcome actor world!";
}

public sealed class HelloActor : ReceiveActor
{
    public sealed record Hello(string Name);
    public sealed record HelloResponse(string Message);

    private readonly IWelcomeMessageProvider _welcomeMessageProvider;

    public HelloActor(IWelcomeMessageProvider welcomeMessageProvider)
    {
        _welcomeMessageProvider = welcomeMessageProvider;

        Receive<Hello>(msg =>
        {
            Sender.Tell(new HelloResponse(_welcomeMessageProvider.GetMessage()));
        });
    }
}

File: skill-test/projects/sample-cluster-java/src/main/java/cluster/java/HelloActor.java
package cluster.java;

import akka.actor.AbstractActor;
import akka.actor.Props;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

@Component("helloActorBean")
@Scope("prototype")
public class HelloActor extends AbstractActor {

    public static Props props(WelcomeMessageProvider welcomeMessageProvider) {
        return Props.create(HelloActor.class, () -> new HelloActor(welcomeMessageProvider));
    }

    public record Hello(String name) {
    }

    public record HelloResponse(String message) {
    }

    private final WelcomeMessageProvider welcomeMessageProvider;

    public HelloActor(WelcomeMessageProvider welcomeMessageProvider) {
        this.welcomeMessageProvider = welcomeMessageProvider;
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
            .match(Hello.class, msg -> getSender().tell(new HelloResponse(welcomeMessageProvider.message()), getSelf()))
            .build();
    }
}
